plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -1015,5 +1030,81 @@ bool GSOSDDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
 	}
 
 
+	return true;
+}
+
+// Debug Configuration Dialog
+
+GSDebugDlg::GSDebugDlg() :
+	GSDialog(IDD_DEBUG)
+{}
+
+void GSDebugDlg::OnInit()
+{
+	HWND hwnd_renderer = GetDlgItem(GetParent(m_hWnd), IDC_RENDERER);
+	GSRendererType renderer = static_cast<GSRendererType>(SendMessage(hwnd_renderer, CB_GETITEMDATA, SendMessage(hwnd_renderer, CB_GETCURSEL, 0, 0), 0));
+
+	bool ogl = renderer == GSRendererType::OGL_HW;
+
+	CheckDlgButton(m_hWnd, IDC_GL_DEBUGLOG, theApp.GetConfigB("debug_opengl"));
+	CheckDlgButton(m_hWnd, IDC_GL_DEBUGGLSL, theApp.GetConfigB("debug_glsl_shader"));
+	CheckDlgButton(m_hWnd, IDC_DUMP_GS, theApp.GetConfigB("dump"));
+	CheckDlgButton(m_hWnd, IDC_DUMP_SAVE, theApp.GetConfigB("save"));
+	CheckDlgButton(m_hWnd, IDC_DUMP_SAVEF, theApp.GetConfigB("savef"));
+	CheckDlgButton(m_hWnd, IDC_DUMP_SAVET, theApp.GetConfigB("savet"));
+	CheckDlgButton(m_hWnd, IDC_DUMP_SAVEZ, theApp.GetConfigB("savez"));
+
+	SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEL), UDM_SETRANGE, 0, MAKELPARAM(10000, 0));
+	SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEL), UDM_SETPOS, 0, MAKELPARAM(theApp.GetConfigB("savel"), 0));
+	SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEN), UDM_SETRANGE, 0, MAKELPARAM(10000, 0));
+	SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEN), UDM_SETPOS, 0, MAKELPARAM(theApp.GetConfigB("saven"), 0));
+
+	EnableWindow(GetDlgItem(m_hWnd, IDC_GL_DEBUGLOG), ogl);
+	EnableWindow(GetDlgItem(m_hWnd, IDC_GL_DEBUGGLSL), ogl);
+
+	UpdateControls();
+}
+
+void GSDebugDlg::UpdateControls() { }
+
+bool GSDebugDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
+{
+	switch (message)
+	{
+	case WM_COMMAND:
+	{
+		int id = LOWORD(wParam);
+
+		switch (id)
+		{
+		case IDOK:
+		{
+			INT_PTR data;
+			theApp.SetConfig("debug_opengl", (int)IsDlgButtonChecked(m_hWnd, IDC_GL_DEBUGLOG));
+			theApp.SetConfig("debug_glsl_shader", (int)IsDlgButtonChecked(m_hWnd, IDC_GL_DEBUGGLSL));
+			theApp.SetConfig("dump", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_GS));
+			theApp.SetConfig("save", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVE));
+			theApp.SetConfig("savef", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVEF));
```

Comment:
Are the casts necessary?

Reply:
I followed the other windows as an example.

Reply:
I'd prefer if redundant casts are removed.